lib/puppet/ssl/host.rb

```diff
@@ -371,27 +378,61 @@ def state
 
   private
 
+  # Ensures that CRL is either on disk, or that CRL checking has been disabled
+  def ensure_crl!
+    if crl_usage && !Puppet::FileSystem.exist?(crl_path)
+      # The CertificateRevocationList indirector will attempt to download the
+      # CRL from the CA if it does not exist on disk.
+      if !CertificateRevocationList.indirection.find(name)
+        Puppet.debug _("Certificate revocation checking is enabled but a CRL cannot be found; CRL checking will not be performed.")
+        @crl_usage = false
```

Comment:
It seems to me that this should be an error...

It also might better to directly set Puppet.setting/Puppet.push_context here rather than only set it on the Host object?

...But really I think this should be an error.